### DISCUSSION TITLE
Endpoint doesn't return json

### DISCUSSION THREAD

**Discussion by mdings** [Budibase/budibase#4622] [General]
For some reason my endpoint return data isn't parsed as json by default. I have tried it in other apps like n8n where the return data is parsed properly. I have also tried explicitly  setting both the Accept:application/json and Content-Type:application/json headers. Currently I have to run `return JSON.parse(data)` in the transformer function to have the data parsed as JSON. I'm wondering why that happens though and why I need to take that extra step? Any thoughts?

**Comment by mjashanks** [comment]
I've found this to often work for me, but sometimes not - depending on the API that I'm using.

Usually, the solution is to add the headers:

`Accept : application/json`
and
`Content-Type: application/json`

I actually think we should set these headers by default.

**Comment by mdings** [comment]
Hi @mjashanks

I have those in my headers, as stated in the original question. However, that also doesn't do the trick, unfortunately.

**Reply by mjashanks** [reply]
apologies - didn't read that properly.

I suspect that the API is returning

`'{ "key": "value" }'` ... i.e. the whole object wrapped in quotes. Thus Budibase thinks it's a string. Other platforms must account for this scenario.
